db/migrate/20190609174303_add_circuit_elements.rb

```diff
@@ -0,0 +1,15 @@
+class AddCircuitElements < ActiveRecord::Migration[5.1]
+  def change
+    create_table :circuit_elements do |t|
+      t.string :name
+      t.string :image
+      t.integer :category
+      t.timestamps
+    end
+
+    create_table :assignments_circuit_elements, id: false do |t|
+      t.belongs_to :circuit_element, index: true
+      t.belongs_to :assignment, index: true
```

Comment:
So, I take it that if there is a record of a circuit element in `assignments_circuit_elements`, the usage of the circuit element will not be allowed. This way be default everything is allowed.

Reply:
Yeah, I was actually thinking of implementing it the other way around but I think this will be better. 😁

Reply:
Good, because then by default nothing should change .. 
Also please keep in mind that as we discussed after the database changes, all changes should happen in the front-end only. The server will not be doing any validations of the same.

Reply:
> The server will not be doing any validations of the same. That is the front-end can verify if there is any violation of the circuit elements usage

I think both the front end and backend should have validations on the circuits elements, so that if the front end validations fail, then there are still there in the backend. In the backend the validations will only be there when the user decides to save the circuits, and maybe when I have to show to the mentor which all restricted elements the student used.